From 4cfba19fa2b087c4b2c5a1b67aa70eb16e9d5a59 Mon Sep 17 00:00:00 2001
From: Laszlo Ersek <lersek@redhat.com>
Date: Wed, 25 May 2022 09:19:58 +0200
Subject: [PATCH] generator/customize: reintroduce "--selinux-relabel" as a
 compat option

Removing "--selinux-relabel" in commit 2f6a27f1077d ("generator/customize:
invert SELinux relabeling default", 2022-05-11) breaks existing scripts
that invoke virt-customize and/or virt-sysprep with that option. Restore
the option, with no functionality tied to it.

Fixes: 2f6a27f1077d32d1ab526427052fc88e188356f7
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2089748
Signed-off-by: Laszlo Ersek <lersek@redhat.com>
Message-Id: <20220525071958.9612-1-lersek@redhat.com>
Reviewed-by: Richard W.M. Jones <rjones@redhat.com>
(cherry picked from commit 4b9ee1052a4396621485fdd56d6826714e7481b1)
---
 generator/customize.ml | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/generator/customize.ml b/generator/customize.ml
index 9634dad85..5abaf206f 100644
--- a/generator/customize.ml
+++ b/generator/customize.ml
@@ -581,6 +581,13 @@ option disables the automatic relabeling.
 The option is a no-op for guests that do not support SELinux.";
   };
 
+  { flag_name = "selinux-relabel";
+    flag_type = FlagBool false;
+    flag_ml_var = "selinux_relabel_ignored";
+    flag_shortdesc = "Compatibility option doing nothing";
+    flag_pod_longdesc = "This is a compatibility option that does nothing.";
+  };
+
   { flag_name = "sm-credentials";
     flag_type = FlagSMCredentials "SELECTOR";
     flag_ml_var = "sm_credentials";
-- 
2.31.1

